src/main/java/use_cases/turn_start_use_case/TurnStartInteractor.java

```diff
@@ -0,0 +1,198 @@
+                    keptIds.add(card.getId());
+                    keptNames.add(card.getName());
+                } else {
+                    burntIds.add(card.getId());
+                    burntNames.add(card.getName());
+                }
+
+            }
+        }
+        for (int i = 0; i < 2; i++) {
+            Card ecard = edeck.draw();
+            val2 = player.addCard(ecard);
+            if (val2){
+                keptIds.add(ecard.getId());
+                keptNames.add(ecard.getName());
+            } else {
+                burntIds.add(ecard.getId());
+                burntNames.add(ecard.getName());
+            }
+
+        }
+
+        DrawCardOutputModel output = new DrawCardOutputModel(keptIds,keptNames,burntIds,burntNames);
+        return turnStartPresenter.showDrawResult(output);
+    }
+
+
+    @Override
+    public CreatureStatsUpdateModel clearBuffs() {
+        Player player = gameState.getCurrentPlayer();
+        List<CreatureCard> creatures = player.getCreatures();
+        for (CreatureCard creature : creatures) {
+            creature.clearDamageBuff();
+            creature.clearHealthBuff();
+            creature.clearStun();
+        }
+
+        CreatureStatsUpdateModel outputData = getCreatureStatsModel();
+        return turnStartPresenter.showClearBuffs(outputData);
+    }
+
+    /**
+     * Trigger the start-of-turn effects in the active structure of the current
+     * player, if it does exist.
+     * This method returns many values because the effects have a variety of
+     * targets, so outcomes also vary.
+     *
+     * @return a TriggerEffectUpdateModel, a combination of a list of hand card
+     * ids and a CreatureStatsUpdateModel.
+     */
+    @Override
+    public TriggerEffectUpdateModel triggerTurnStartEffects() {
+        Player player1 = gameState.getCurrentPlayer();
+        Player player2 = gameState.getOpposingPlayer();
+        if (player1.getStructure() != null) {
+            StructureCard structureCard = player1.getStructure();
+            List<CreatureCard> effectTargets = new ArrayList<>();
+            Player effectplayer = null;
+            if (structureCard.getTriggerEvent() == GameEvent.TURN_START) {
+                switch (structureCard.getTargetType()) {
+                    case ALL:
+                        effectTargets.addAll(player1.getCreatures());
+                        effectTargets.addAll(player2.getCreatures());
+                        break;
+                    case FRIENDLY:
+                        effectTargets.addAll(player1.getCreatures());
+                        break;
+                    case ENEMY:
+                        effectTargets.addAll(player2.getCreatures());
+                        break;
+                    case OPPONENT:
+                        effectplayer = player2;
+                        break;
+                    case SELF:
+                        effectplayer = player1;
+                        break;
```

Comment:
there is actually one structure that draws card for the player, check the json files in utils (well I guess my branch hasnt been merged so just check the files in that pull request for now.)

also adding self and opponent would cover all possible situations, like if we do add a structure card that addresses the players, we dont have to change this code (although we're probably still changing this code later because there's probably better ways to implement it)